lib/spdy-transport/connection.js

```diff
@@ -284,11 +307,26 @@ Connection.prototype._handleFrame = function _handleFrame(frame) {
         return;
 
       state.debug('id=0 stream=%d not found', frame.id);
-      state.framer.rstFrame({ id: frame.id, code: 'INVALID_STREAM' });
+      this._goaway({
+        lastId: state.stream.lastId.both,
+        code: 'PROTOCOL_ERROR',
+        send: true
+      });
       return;
     }
   }
 
+  if (!stream && state.isServer && state.version === 4 && frame.id !== 0 &&
+      (frame.id % 2 === 0 || frame.id < state.stream.lastId.received)) {
```

Comment:
It should be fine to leave these two conditions on the same line, though, as they are in parens.